src/homekit.cs

```diff
@@ -1492,4 +1629,165 @@ interface HMLocationEvent {
 		[Export ("updateRegion:completionHandler:")]
 		void UpdateRegion (CLRegion region, Action<NSError> completion);
 	}
+
+	[NoWatch, TV (10,0), iOS (10,0)]
+	[BaseType (typeof(UIView))]
+	interface HMCameraView
+	{
+		[NullAllowed, Export ("cameraSource", ArgumentSemantic.Strong)]
+		HMCameraSource CameraSource { get; set; }
+	}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(NSObject))]
+	interface HMCameraSource {}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(HMAccessoryProfile))]
+	[DisableDefaultCtor]
+	interface HMCameraProfile
+	{
+		[NullAllowed, Export ("streamControl", ArgumentSemantic.Strong)]
+		HMCameraStreamControl StreamControl { get; }
+
+		[NullAllowed, Export ("snapshotControl", ArgumentSemantic.Strong)]
+		HMCameraSnapshotControl SnapshotControl { get; }
+
+		[NullAllowed, Export ("settingsControl", ArgumentSemantic.Strong)]
+		HMCameraSettingsControl SettingsControl { get; }
+
+		[NullAllowed, Export ("speakerControl", ArgumentSemantic.Strong)]
+		HMCameraAudioControl SpeakerControl { get; }
+
+		[NullAllowed, Export ("microphoneControl", ArgumentSemantic.Strong)]
+		HMCameraAudioControl MicrophoneControl { get; }
+	}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(NSObject))]
+	interface HMCameraControl {}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(HMCameraControl))]
+	interface HMCameraStreamControl
+	{
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		HMCameraStreamControlDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("streamState", ArgumentSemantic.Assign)]
+		HMCameraStreamState StreamState { get; }
+
+		[NullAllowed, Export ("cameraStream", ArgumentSemantic.Strong)]
+		HMCameraStream CameraStream { get; }
+
+		[Export ("startStream")]
+		void StartStream ();
+
+		[Export ("stopStream")]
+		void StopStream ();
+	}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface HMCameraStreamControlDelegate
+	{
+		[Export ("cameraStreamControlDidStartStream:")]
+		void DidStartStream (HMCameraStreamControl cameraStreamControl);
+
+		[Export ("cameraStreamControl:didStopStreamWithError:")]
+		void DidStopStream (HMCameraStreamControl cameraStreamControl, NSError error);
+	}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(HMCameraSource))]
+	interface HMCameraStream
+	{
+		[Export ("audioStreamSetting", ArgumentSemantic.Assign)]
+		HMCameraAudioStreamSetting AudioStreamSetting { get; set; }
+	}
+
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(HMCameraControl))]
+	interface HMCameraSnapshotControl
+	{
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		HMCameraSnapshotControlDelegate Delegate { get; set; }
```

Comment:
`HMCameraSnapshotControlDelegate` will be the model, not the protocol interface (forcing the use of the model).